waku/v2/api/missing/missing_messages.go

```diff
@@ -123,24 +140,33 @@ func (m *MissingMessageVerifier) Start(ctx context.Context) {
 				for _, interest := range critIntList {
 					select {
 					case <-ctx.Done():
+						m.setRunning(false)
 						return
 					default:
 						semaphore <- struct{}{}
 						go func(interest criteriaInterest) {
 							defer utils.LogOnPanic()
-							m.fetchHistory(c, interest)
+							m.fetchHistory(m.C, interest)
 							<-semaphore
 						}(interest)
 					}
 				}
 
 			case <-ctx.Done():
+				m.setRunning(false)
 				return
 			}
 		}
 	}()
 }
 
+func (m *MissingMessageVerifier) Stop() {
+	m.cancel()
+	m.runningMutex.Lock()
+	defer m.runningMutex.Unlock()
+	m.isRunning = false
```

Comment:
Do you think we should recreate the channel `C` on `Stop()` to reset it?

Reply:
Valid point, but the caller also needs to keep recreating the routine to listen to channel every time we recreate which seemed like it would complicate things. hence i left it as is. 

Any specific issue you see if we don't recreate the channel?

Reply:
No specific issue, just curious.
I just find that losing the channel after a stop seemed logical as it kinda notifies all subscribers to it that they're no longer going to receive messages, but it's totally okay to leave the channel alive.